src/bootstrap/src/core/config/config.rs

```diff
@@ -325,6 +325,9 @@ pub struct Config {
     pub hosts: Vec<TargetSelection>,
     pub targets: Vec<TargetSelection>,
     pub local_rebuild: bool,
+    #[cfg(not(test))]
```

Comment:
Is this really needed? We also have other things configured per target (e.g. codegen-backends) and these are also `pub`.

Reply:
For fields meant to be accessed only through the relevant method, I think it makes sense to make them private. This way it's not possible for future PRs to accidentally access `config.jemalloc` directly.

Reply:
Ok, we already do the same for `initial_rustfmt`, didn't notice.